includes/class-fs-api.php

```diff
@@ -318,9 +318,12 @@ function get( $path = '/', $flush = false, $expiration = WP_FS__TIME_24_HOURS_IN
 				$flush = true;
 			}
 
-			$cached_result = self::$_cache->get( $cache_key );
+            $has_valid_cache = self::$_cache->has_valid( $cache_key, $expiration );
+            $cached_result   = $has_valid_cache ?
+                self::$_cache->get( $cache_key ) :
+                null;
 
-			if ( $flush || ! self::$_cache->has_valid( $cache_key, $expiration ) ) {
+            if ( $flush || ! $has_valid_cache ) {

```

Comment:
I suggest to do it like

```php
if ( $flush || is_null( $cached_result ) {
```

This is just for self-guarding code, where the `$_cache->get` would return `null` if not present.